### PR TITLE
Pull Request for Issue 80 and Issue 82

### PR DESCRIPTION
**Pull request by Alexhuszagh**
I've updated the license to use the FreeBSD license, and so this should be ready to merge.

### CODE REVIEW

**Comment by jmcnamara** [comment]
Hi @Alexhuszagh,

Some minor comments.

* The title of the pull request and commit message should say what the commit does. The title shouldn't refer to the issue numbers it addresses. Something like:

  ```
   Add CMake build support.

   Added support for a CMake build system.
  ```
* The pull request needs to be squashed to a single commit.
* The `const char*` doesn't belong in this patchset. You should open a new bug report issue for that. A PR isn't required.

Thanks,

John

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+#   Use:
+#       Move to a custom directory, ideally out of source, and
+#       type `cmake $LXW_SOURCE $FLAGS`, where `LXW_SOURCE` is the
+#       path to the libxlsxwriter project, and `FLAGS` are custom
+#       flags to pass to the compiler.
+#
+#   Example:
+#       For example, in the build directory, to build libxlsxwriter
+#       and the unittests in debug mode, type:
+#           cmake .. -DBUILD_TESTS=ON
```
Comment (jmcnamara) [inline CMakeLists.txt]: Probably should add `cd cmake` as well.

File: CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+#           https://cmake.org/cmake/help/v3.0/manual/cmake-generators.7.html
+#
+
+cmake_minimum_required(VERSION 2.8)
+project(xlsxwriter C)
+
+# VERSION
+# -------
+
+set(LXW_VERSION include/xlsxwriter.h)
```
Comment (jmcnamara) [inline CMakeLists.txt]: Is this versioning used by a general user of CMake. If not leave it out.

**Comment by Alexhuszagh** [comment]
Hi @jmcnamara, sorry, some code from upstream got added into the current patch. I'll patch that shortly, that code has no place in here. I'll make a diff, revert and fix the comments, and then update the pull request.

File: CMakeLists.txt (hunk 2 above)
Comment (Alexhuszagh) [inline CMakeLists.txt]: It is only used for packaging, I can leave it out.

**Comment by jmcnamara** [comment]
Merged. Thanks.

**Pull request merged**